app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partnership/PartnershipOtherPartnerContactNameController.scala

```diff
@@ -0,0 +1,152 @@
+        organisationName <- inflight.organisationName
+      } yield (for {
+        contactDetails <- inflight.contactDetails
+        firstName      <- contactDetails.firstName
+        lastName       <- contactDetails.lastName
+
+      } yield (firstName, lastName)) match {
+        case Some(data) =>
+          Ok(
+            page(MemberName.form().fill(MemberName(data._1, data._2)),
+                 organisationName,
+                 partnershipRoutes.PartnershipPartnersListController.displayPage(),
+                 partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
+            )
+          )
+        case None =>
+          Ok(
+            page(MemberName.form(),
+                 organisationName,
+                 partnershipRoutes.PartnershipPartnersListController.displayPage(),
+                 partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
+            )
+          )
+      }).getOrElse(NotFound)
+    }
+
+  def submit(): Action[AnyContent] =
+    (authenticate andThen journeyAction).async { implicit request =>
+      val inflightOtherPartner =
+        Some(
+          Partner(organisationName = Some("Partner organisation"),
+                  partnerType = Some(PartnerTypeEnum.SOLE_TRADER)
+          )
+        ) // TODO will have been populated by the preceding GRS journey
+
+      (for {
+        inflight         <- inflightOtherPartner
+        organisationName <- inflight.organisationName
+      } yield MemberName.form()
+        .bindFromRequest()
+        .fold(
+          (formWithErrors: Form[MemberName]) =>
+            Future.successful(
+              BadRequest(
+                page(formWithErrors,
+                     organisationName,
+                     partnershipRoutes.PartnershipPartnersListController.displayPage(),
+                     partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
+                )
+              )
+            ),
+          fullName =>
+            updateRegistration(fullName).map {
+              case Right(_) =>
+                FormAction.bindFromRequest match {
+                  case SaveAndContinue =>
+                    Redirect(routes.PartnershipOtherPartnerEmailAddressController.displayPage())
+                  case _ =>
+                    Redirect(
+                      partnershipRoutes.PartnershipOtherPartnerContactNameController.displayPage()
+                    )
+                }
+              case Left(error) => throw error
+            }
+        )).getOrElse(Future.successful(NotFound))
+    }
+
+  private def updateRegistration(
+    formData: MemberName
+  )(implicit req: JourneyRequest[AnyContent]): Future[Either[ServiceError, Registration]] =
+    update { registration =>
+      val otherPartner = registration.inflightPartner.getOrElse(
+        Partner(partnerType = Some(PartnerTypeEnum.SOLE_TRADER))
+      ) // TODO type from previous GRS screen
```

Comment:
This creation of the Partner will be removed once the GRS screen is in front of us.